File: youlexuan_sellergoods_interface/src/main/java/com/offcn/sellergoods/service/BrandService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.entity.Result;
import com.offcn.pojo.TbBrand;
import com.sun.javafx.logging.PulseLogger;

import java.util.List;
import java.util.Map;

public interface BrandService {

    public List<TbBrand> findAll();

    public PageResult findPage(int pageNum,int pageSize);

    public void add(TbBrand brand);

    public void update(TbBrand brand);

    public void delete(Long[] ids);

    public TbBrand selectOne(Long id);

    public PageResult findPage(int pageNum,int pageSize,TbBrand brand);

    public List<Map> selectOptionList();



}

File: youlexuan_manager_web/src/main/webapp/js/service/brandService.js
//定义一个服务层
//参数一:服务的名字
//参数二:服务干的事情
app.service('brandService',function ($http) {
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    }

    this.findPage = function (page,size) {
        return $http.get('../brand/findPage.do?page=' + page + '&size=' + size);
    }

    this.search = function (page,size,searchEntity) {
        return $http.post('../brand/search.do?page='+page+'&size='+size,searchEntity);
    }

    this.save = function (entity) {
        /*新增*/
        var methodName='add';
        if(entity.id != null){
            /*修改*/
            methodName = 'update';
        }
        return $http.post('../brand/'+methodName+'.do',entity);
    }

    this.findOne = function (id) {
        return $http.get('../brand/selectOne?id='+id);
    }

    this.delete = function (selectIds) {
        return $http.get('../brand/delete.do?ids='+selectIds);
    }

    /*获取品牌的id和name*/
    this.selectOptionList = function () {
        return $http.get('../brand/selectOptionList.do');
    }

})